### PR TITLE
Calcite radiodelegate

### PR DESCRIPTION
**Pull request by mrwillyees**

### CODE REVIEW

File: calcite-qml/Calcite/RadioDelegate.qml [hunk 1]
```diff
@@ -15,34 +16,62 @@
...
 
 T.RadioDelegate {
     id: control
 
-    implicitWidth: Math.max(background ? background.implicitWidth : 0,
-                            contentItem.implicitWidth + leftPadding + rightPadding)
-    implicitHeight: Math.max(background ? background.implicitHeight : 0,
-                             Math.max(contentItem.implicitHeight,
-                                      indicator ? indicator.implicitHeight : 0) + topPadding + bottomPadding)
+    Component.onCompleted: console.log(width)
```
Comment (anmacdonald) [inline calcite-qml/Calcite/RadioDelegate.qml:24]: ❌ Remove

File: uitools/images/esri_arcgisruntime_toolkit_images.qrc [hunk 2]
```diff
@@ -21,5 +21,8 @@
         <file>text_editing_mode.png</file>
         <file>x.svg</file>
         <file>zoom.png</file>
+        <file>chevrons-right.svg</file>
+        <file>zoom-to-object.svg</file>
+        <file>chevrons-left.svg</file>
```
Comment (anmacdonald) [inline uitools/images/esri_arcgisruntime_toolkit_images.qrc:26]: 🙃  Alphabetical order.

File: calcite-qml/Calcite/RadioDelegate.qml [hunk 3]
```diff
@@ -15,34 +16,62 @@
...
+        width: icon.width
+        height: icon.height
+        ColorOverlay {
+            anchors.fill: indicator
+            source: indicator
+            color: control.checked
+                   || control.highlighted ? Calcite.brand : control.hovered ? Calcite.border1 : "transparent"
+            visible: indicator.visible
+        }
+    }
```
Comment (anmacdonald) [inline calcite-qml/Calcite/RadioDelegate.qml:69]: ❌ (This is my fault by taking the initial implementation from an older Material implementation.) I'd just like to quickly compare to how Material's RadioIndicator works as of Qt 5.15.7, and how it interacts with `icon`:

* Materials uses an `IconLabel` which takes an icon property. Here we are using   a plain `Text`.
* Material's `indicator` is independent of the icon. Here `icon` and `indicator` are linked.

I believe that you should take the same approach as Material. The `icon` should be an optional item that can be added to the `RadioDelegate` to show an image. If the user wants to change the indicator they should override the `indicator` property, not the `icon` property.

Material implementation below:

```qml
    indicator: RadioIndicator {
        x: control.text ? (control.mirrored ? control.leftPadding : control.width - width - control.rightPadding) : control.leftPadding + (control.availableWidth - width) / 2
        y: control.topPadding + (control.availableHeight - height) / 2
        control: control
    }

    contentItem: IconLabel {
        leftPadding: !control.mirrored ? 0 : control.indicator.width + control.spacing
        rightPadding: control.mirrored ? 0 : control.indicator.width + control.spacing

        spacing: control.spacing
        mirrored: control.mirrored
        display: control.display
        alignment: control.display === IconLabel.IconOnly || control.display === IconLabel.TextUnderIcon ? Qt.AlignCenter : Qt.AlignLeft

        icon: control.icon
        text: control.text
        font: control.font
        color: control.enabled ? control.Material.foreground : control.Material.hintTextColor
    }

    background: Rectangle {
        implicitHeight: control.Material.delegateHeight

        color: control.highlighted ? control.Material.listHighlightColor : "transparent"

        Ripple {
            width: parent.width
            height: parent.height

            clip: visible
            pressed: control.pressed
            anchor: control
            active: control.down || control.visualFocus || control.hovered
            color: control.Material.rippleColor
        }
    }
```

File: calcite-qml/Calcite/Calcite.qml [hunk 4]
```diff
@@ -43,18 +43,28 @@ QtObject {
...
+    readonly property color foreground5: theme === Calcite.Theme.Light ? "#EAEAEA" : "#2B2B2B"
+
     readonly property color text1: theme === Calcite.Theme.Light ? "#151515" : "#FFFFFF"
 
     readonly property color text2: theme === Calcite.Theme.Light ? "#4A4A4A" : "#BFBFBF"
 
     readonly property color text3: theme === Calcite.Theme.Light ? "#6A6A6A" : "#9F9F9F"
 
+    readonly property color text4: theme === Calcite.Theme.Light ? "#2B2B2B" : "#FFFFFF"
+
```
Comment (anmacdonald) [inline calcite-qml/Calcite/Calcite.qml:67]: 🔧 As mentioned in the previously PR. `background2`, `background3`, `foreground4`, `foreground5`, and `text4` are not part of the [base](https://developers.arcgis.com/calcite-design-system/foundations/colors/#themes) properties [available](https://github.com/Esri/calcite-colors/blob/master/src/colors.ts#L832) in other Calcite definitions. Please consider making the RadioButton use the already available color properties, updating the available colors if required.

File: calcite-qml/Calcite/RadioDelegate.qml (hunk 3 above)
Comment (anmacdonald) [inline calcite-qml/Calcite/RadioDelegate.qml:69]: 🔧  Other calcite controls change the opacity when disabled , you might want to test checking/unchecking the enabled box in the calcite demo app to ensure the component "looks" disabled when required.

image

**Review by anmacdonald — CHANGES_REQUESTED** [review]
👍 Looking snazzy! Some thoughts provided on indicator and coloring.

File: calcite-qml/Calcite/RadioDelegate.qml (hunk 3 above)
Comment (mrwillyees) [reply calcite-qml/Calcite/RadioDelegate.qml:69]: Ok, by using an `IconLabel` we can follow even more closely the Calcite standard by having an extra image at the beginning of the component (`icon`). I am using the `indicator` to show the bullet point as default.
image

File: calcite-qml/Calcite/RadioDelegate.qml [hunk 5]
```diff
@@ -15,34 +16,69 @@
...
     id: control
 
-    implicitWidth: Math.max(background ? background.implicitWidth : 0,
-                            contentItem.implicitWidth + leftPadding + rightPadding)
-    implicitHeight: Math.max(background ? background.implicitHeight : 0,
-                             Math.max(contentItem.implicitHeight,
-                                      indicator ? indicator.implicitHeight : 0) + topPadding + bottomPadding)
+    // extra colors not part of calcite specification
+    readonly property int theme: Calcite.theme
+    readonly property color backgroundDown: theme === Calcite.Theme.Light ? "#C7EAFF" : "#151515"
```
Comment (anmacdonald) [inline calcite-qml/Calcite/RadioDelegate.qml:28]: 🔧 Are these colors able to be simulated from what is currently available?

**Review by anmacdonald — APPROVED** [review]
LGTM, one final query!

**Pull request merged**